test/local/senders/sms.js

```diff
@@ -4,77 +4,183 @@
 
 'use strict'
 
-const ROOT_DIR = '../../..'
-
 const assert = require('insist')
+const mocks = require('../../mocks')
 const P = require('bluebird')
 const proxyquire = require('proxyquire')
 const sinon = require('sinon')
 
-const log = {
-  error: sinon.spy(),
-  info: sinon.spy(),
-  trace: sinon.spy()
-}
-
-let snsResult = P.resolve({
-  MessageId: 'foo'
-})
-const publish = sinon.spy(params => ({
-  promise: () => snsResult
-}))
-function SNS () {}
-SNS.prototype.publish = publish
-
-let mockConstructed = false
-function MockSNS () {
-  mockConstructed = true
-}
-MockSNS.prototype = SNS.prototype
```

Comment:
All this global state and the need to call `reset` inside `afterEach` was annoying me, hence the move to constructing stuff in `beforeEach` and the various changes that induced.